rust/codelist-rs/src/codelist.rs

```diff
@@ -67,38 +83,306 @@ impl CodeList {
         }
     }
 
+    /// Get the type of the codelist
+    ///
+    /// # Returns
+    /// * `&CodeListType` - The type of the codelist
     pub fn codelist_type(&self) -> &CodeListType {
         &self.codelist_type
     }
 
+    /// Add an entry to the codelist
+    ///
+    /// # Arguments
+    /// * `code` - The code to add
+    /// * `term` - The term to add
     pub fn add_entry(&mut self, code: String, term: String) {
         let entry = CodeEntry::new(code, term);
         self.entries.insert(entry);
     }
 
-    pub fn remove_entry(&mut self, code: &str) {
-        // Some implementation
+    /// Remove an entry from the codelist
+    ///
+    /// # Arguments
+    /// * `code` - The code to remove
+    ///
+    /// # Errors
+    /// * `CodeListError::EntryNotFound` - If the entry to be removed is not found
+    pub fn remove_entry(&mut self, code: &str) -> Result<(), CodeListError> {
+        let initial_size = self.entries.len();
+        self.entries.retain(|entry| entry.code != code);
+        let final_size = self.entries.len();
+        if initial_size == final_size {
+            return Err(CodeListError::EntryNotFound(code.to_string()));
+        }
```

Comment:
There might be an easier way to do this. I think you can just use `.remove` as this is a Hashset (https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.remove). 

This should evaulate to true or false, and we can use this to check for error. 

Something like;

```rust
pub fn remove_entry(&mut self, code: &str) -> Result<(), CodeListError> {
    let removed = self.entries.remove(&CodeEntry::new(code.to_string(), String::new()));
    if removed { // this is checking if true
        Ok(())
    } else {
        Err(CodeListError:...etc
    }
}
```